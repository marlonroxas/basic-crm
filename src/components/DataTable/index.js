import React from "react";
import moment from 'moment'
import { EditOutlined, DeleteOutlined, EyeOutlined } from '@ant-design/icons'
import SortIcon from "../../img/sort-icon.png";
import "./style.scss";

/* Data list view */
const DataTable = props => {
  return (
    <div className="table-wrapper">
      <table className="data-table">
        <thead>
          <tr>
            <th
              onClick={() => {
                props.onSortChange("id");
              }}
            >
              <span className="column-sort">
                Code
              </span>
            </th>
            <th
              onClick={() => {
                props.onSortChange("name");
              }}
            >
              <span className="column-sort">
                First Name
                <img src={SortIcon} alt="First Name" />
              </span>
            </th>
            <th
              onClick={() => {
                props.onSortChange("surname");
              }}
            >
              <span className="column-sort">
                Last Name
                <img src={SortIcon} alt="Last Name" />
              </span>
            </th>
            <th
              onClick={() => {
                props.onSortChange("birthdate");
              }}
            >
              <span className="column-sort">
                Birthdate
              </span>
            </th>
            <th
              onClick={() => {
                props.onSortChange("email");
              }}
            >
              <span className="column-sort">
                Email
              </span>
            </th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          {props.users && props.users.length ? (
            props.users.map(user => (
              <tr key={user.id}>
                <td>{(`${user.first_name}${user.last_name}${user.birthdate ? moment(user.birthdate).format('yyyyMMDD') : ''}`).replace(/ /g, '').toLocaleLowerCase()}</td>
                <td>{user.first_name}</td>
                <td>{user.last_name}</td>
                <td>{user.birthdate}</td>
                <td>{user.email}</td>
                <td className="field-actions">
                <button
                    className="field-actions__view"
                    onClick={() => {
                      props.viewDetails(user);
                    }}
                    alt="View details"
                    title="View details"
                  >
                    <EyeOutlined />
                  </button>
                  <button
                    className="field-actions__update"
                    onClick={() => {
                      props.updateRow(user);
                    }}
                    alt="Update details"
                    title="Update details"
                  >
                    <EditOutlined />
                  </button>
                  <button
                    className="field-actions__delete"
                    onClick={() => props.deleteRow(user)}
                    alt="Delete this user"
                    title="Delete this user"
                  >
                    <DeleteOutlined />
                  </button>
                </td>
              </tr>
            ))
          ) : (
            <tr>
              <td colSpan="5">
                <div className="no-record-message">No Record!</div>
              </td>
            </tr>
          )}
        </tbody>
      </table>
    </div>
  );
};

export default DataTable;
