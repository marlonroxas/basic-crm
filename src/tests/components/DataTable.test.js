import React from 'react'
import { shallow } from 'enzyme'

import DataTable from '../../components/DataTable/index'

test('Should render DataTable correctly', () => {
  const wrapper = shallow(<DataTable />)
  expect(wrapper).toMatchSnapshot()
})