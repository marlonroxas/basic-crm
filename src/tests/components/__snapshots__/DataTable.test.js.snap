// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DataTable correctly 1`] = `
<div
  className="table-wrapper"
>
  <table
    className="data-table"
  >
    <thead>
      <tr>
        <th
          onClick={[Function]}
        >
          <span
            className="column-sort"
          >
            Code
          </span>
        </th>
        <th
          onClick={[Function]}
        >
          <span
            className="column-sort"
          >
            First Name
            <img
              alt="First Name"
              src="sort-icon.png"
            />
          </span>
        </th>
        <th
          onClick={[Function]}
        >
          <span
            className="column-sort"
          >
            Last Name
            <img
              alt="Last Name"
              src="sort-icon.png"
            />
          </span>
        </th>
        <th
          onClick={[Function]}
        >
          <span
            className="column-sort"
          >
            Birthdate
          </span>
        </th>
        <th
          onClick={[Function]}
        >
          <span
            className="column-sort"
          >
            Email
          </span>
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="5"
        >
          <div
            className="no-record-message"
          >
            No Record!
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
